File: LeetCode 热题 HOT 100/二叉树/101. 对称二叉树（层次遍历）.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
from collections import deque


class Solution:
    def isSymmetric(root):
        if not root:
            return True

        points = deque([root])
        while points:
            res = []
            size = len(points)
            for _ in range(size):
                temp = points.popleft()
                if temp == -999:
                    res.append(-999)
                else:
                    res.append(temp.val)
                    if temp.left:
                        points.append(temp.left)
                    else:
                        points.append(-999)
                    if temp.right:
                        points.append(temp.right)
                    else:
                        points.append(-999)
            if res != res[::-1]:
                return False
        return True

File: LeetCode 热题 HOT 100/二叉树/101. 对称二叉树（层次遍历）.js
// 给你一个二叉树的根节点 root ， 检查它是否轴对称。 

// 示例 1：
// 输入：root = [1, 2, 2, 3, 4, 4, 3]
// 输出：true

// 示例 2：
// 输入：root = [1, 2, 2, null, 3, null, 3]
// 输出：false


/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isSymmetric = function (root) {
  if (root === null) return true

  let points = []
  points.push(root)
  while (points.length !== 0) {
    let res = []
    let size = points.length
    for (let index = 0; index < size; index++) {
      let temp = points.shift()
      if (temp === -999) {
        res.push(-999)
      } else {
        res.push(temp.val)
        if (temp.left) {
          points.push(temp.left)
        } else {
          points.push(-999)
        }
        if (temp.right) {
          points.push(temp.right)
        } else {
          points.push(-999)
        }
      }
    }
    if (res !== res.reverse()) return false
  }
  return true
};
